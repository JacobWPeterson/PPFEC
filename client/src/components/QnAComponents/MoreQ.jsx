import React from 'react';
import styled from 'styled-components';

const MoreAQ = styled.button`
  background-color: #FFFFFF;
  border: 1px 1px 1px 1px;
  border-color: #424242;
  height: 6vh;
  width: 20vw;
  font-size: 15px;
`;

const MQC = styled.div`
padding: 1vh 0 1vh 0;
`;

class MoreQ extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.func();
  }

  render() {
    return (
      <MQC>
        {this.props.buttonDisplay
          ? <MoreAQ type="submit" onClick={this.handleClick}>MORE ANSWERED QUESTIONS</MoreAQ> : null}
      </MQC>
    );
  }
}

export default MoreQ;
