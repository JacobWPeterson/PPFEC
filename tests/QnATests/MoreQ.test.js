import React from 'react';
import { mount } from 'enzyme';
import MoreQ from '../../client/src/components/QnAComponents/MoreQ.jsx';

describe('This is for the More Questions Component', () => {
  const moreQComp = mount(<MoreQ />);

  it('Should render More Questions Properly', () => {
    expect(moreQComp).toMatchSnapshot();
  });
});
