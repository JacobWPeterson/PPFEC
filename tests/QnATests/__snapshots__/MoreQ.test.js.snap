// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the More Questions Component Should render More Questions Properly 1`] = `
<MoreQ>
  <MoreQ__MQC>
    <div
      className="MoreQ__MQC-ir99cd-1 fxtzCf"
    />
  </MoreQ__MQC>
</MoreQ>
`;

exports[`This is for the More Questions Component Should render More Questions Properly 1`] = `
<MoreQ>
  <MoreQ__MQC>
    <div
      className="MoreQ__MQC-ir99cd-1 fxtzCf"
    />
  </MoreQ__MQC>
</MoreQ>
`;
